src/arkode/arkode_sprkstep.c

```diff
@@ -0,0 +1,781 @@
+                    MSG_ARK_NULL_SUNCTX);
+    return (NULL);
+  }
+
+  /* Test if all required vector operations are implemented */
+  nvectorOK = sprkStep_CheckNVector(y0);
+  if (!nvectorOK)
+  {
+    arkProcessError(NULL, ARK_ILL_INPUT, "ARKODE::SPRKStep", "SPRKStepCreate",
+                    MSG_ARK_BAD_NVECTOR);
+    return (NULL);
+  }
+
+  /* Create ark_mem structure and set default values */
+  ark_mem = arkCreate(sunctx);
+  if (ark_mem == NULL)
+  {
+    arkProcessError(NULL, ARK_MEM_NULL, "ARKODE::SPRKStep", "SPRKStepCreate",
+                    MSG_ARK_NO_MEM);
+    return (NULL);
+  }
+
+  /* Allocate ARKodeSPRKStepMem structure, and initialize to zero */
+  step_mem = NULL;
+  step_mem = (ARKodeSPRKStepMem)malloc(sizeof(struct ARKodeSPRKStepMemRec));
+  if (step_mem == NULL)
+  {
+    arkProcessError(ark_mem, ARK_MEM_FAIL, "ARKODE::SPRKStep", "SPRKStepCreate",
+                    MSG_ARK_ARKMEM_FAIL);
+    return (NULL);
+  }
+  memset(step_mem, 0, sizeof(struct ARKodeSPRKStepMemRec));
+
+  /* Allocate vectors in stepper mem */
+  if (!arkAllocVec(ark_mem, y0, &(step_mem->sdata)))
+  {
+    SPRKStepFree((void**)&ark_mem);
+    return (NULL);
+  }
+
+  /* Attach step_mem structure and function pointers to ark_mem */
+  ark_mem->step_attachlinsol   = NULL;
+  ark_mem->step_attachmasssol  = NULL;
+  ark_mem->step_disablelsetup  = NULL;
+  ark_mem->step_disablemsetup  = NULL;
+  ark_mem->step_getlinmem      = NULL;
+  ark_mem->step_getmassmem     = NULL;
+  ark_mem->step_getimplicitrhs = NULL;
+  ark_mem->step_mmult          = NULL;
+  ark_mem->step_getgammas      = NULL;
+  ark_mem->step_init           = sprkStep_Init;
+  ark_mem->step_fullrhs        = sprkStep_FullRHS;
+  ark_mem->step                = sprkStep_TakeStep;
+  ark_mem->step_mem            = (void*)step_mem;
+
+  /* Set default values for SPRKStep optional inputs */
+  retval = SPRKStepSetDefaults((void*)ark_mem);
+  if (retval != ARK_SUCCESS)
+  {
+    arkProcessError(ark_mem, retval, "ARKODE::SPRKStep", "SPRKStepCreate",
+                    "Error setting default solver options");
+    SPRKStepFree((void**)&ark_mem);
+    return (NULL);
+  }
+
+  /* Copy the input parameters into ARKODE state */
+  step_mem->f1 = f1;
+  step_mem->f2 = f2;
+
+  /* Initialize the counters */
+  step_mem->nf1    = 0;
+  step_mem->nf2    = 0;
+  step_mem->istage = 0;
+
+  /* Initialize main ARKODE infrastructure */
+  retval = arkInit(ark_mem, t0, y0, FIRST_INIT);
+  if (retval != ARK_SUCCESS)
+  {
+    arkProcessError(ark_mem, retval, "ARKODE::SPRKStep", "SPRKStepCreate",
+                    "Unable to initialize main ARKODE infrastructure");
+    SPRKStepFree((void**)&ark_mem);
+    return (NULL);
+  }
+
+  /* We use Lagrange interpolation by default otherwise extra RHS calls are
+     needed. This is because we cannot reuse the f2 RHS in TakeStep since it is
+     a staggered time step. Additionally, it seems Lagrange interpolation does
+     a better job of conservation. */
```

Comment:
I'm not sure that we need so much of a justification for this default here.  Perhaps something simpler:
```suggestion
  /* SPRKStep uses Lagrange interpolation by default, since Hermite is less compatible with these methods. */
```